### DISCUSSION TITLE
Allow local mutation

### DISCUSSION THREAD

**Discussion by user72356** [jonaskello/eslint-plugin-functional#358] [Q&A]
Is there a rule that lets me mutate a local object? Something like:

```js
function test()
{
    const myVar = { hello: "" };
    myVar.hello = "world";
}
```

I feel that maybe there should be an allowLocalMutation rule that allows the code above, for those who want to be less strict about total immutability and only want the input parameters to be immutable?

**Comment by RebeccaStevens** [comment]
What error are you currently getting? Is it for the "no-let" or "immutable-data" rule?

**Comment by user72356** [comment]
I get ```'myVar' is never reassigned. Use 'const' instead.eslint[prefer-const]``` for the let myVar
(Actually for that one technically it's not coming from functional and I should have used a const anyway- Updated OP)

And a ```Modifying an existing object/array is not allowed.eslint[functional/immutable-data]``` under 2nd line.

**Comment by RebeccaStevens** [comment]
I'd recommended setting the option "ignoreAccessorPattern" for "immutable-data".
Then if you rename `myVar` to something like `mutableMyVar` then "immutable-data" will no longer report an issue.

If you aren't reassigning the variable then you should use `const` instead of `let` for the first line.
